### PR TITLE
Add comments & definitions to map_header.h and map_header.c

### PR DESCRIPTION
**Pull request by Mr-Toodguy**

### CODE REVIEW

**Review by lhearachel — CHANGES_REQUESTED** [review]
This is a little hard to review due to the overall diff size of `map_headers.c`. Could we pull the `mapHeaders` table out into a separate file? e.g. `include/data/map_headers.h`, then include the file in `map_headers.c` to bring the total diff size down. While we're at it, let's update the name to be `sMapHeaders`.

**Comment by Mr-Toodguy** [comment]
Ok sounds good, I agree that the giant table in map_headers.c is undesirable.

File: src/map_header.c [hunk 1]
```diff
@@ -2,13059 +2,11 @@
...
-		0xC,
-		0x3,
-		0xA,
-		0x1,
-		0x1,
-		0x1,
-		0x0,
-	},
-};
+#include "data/map_headers.h"
```
Comment (lhearachel) [inline src/map_header.c:5]: :raised_hands:

**Review by lhearachel — APPROVED** [review]
Love to see it.

**Pull request merged**